cmd/shell.go

```diff
@@ -209,42 +209,49 @@ func FailOnError(err error, msg string) {
 func AmqpChannel(conf Config) (*amqp.Channel, error) {
 	var conn *amqp.Connection
 
-	if conf.AMQP.SSL != nil {
-		if strings.HasPrefix(conf.AMQP.Server, "amqps") == false {
-			err := fmt.Errorf("SSL configuration provided, but not using an AMQPS URL")
+	if conf.AMQP.TLS == nil {
+		// Configuration did not specify SSL options
+		conn, err := amqp.Dial(conf.AMQP.Server)
+		if err != nil {
 			return nil, err
 		}
 
-		cfg := new(tls.Config)
+		return conn.Channel()
+	}
 
-		cert, err := tls.LoadX509KeyPair(conf.AMQP.SSL.CertFile, conf.AMQP.SSL.KeyFile)
-		if err != nil {
-			err = fmt.Errorf("Could not load Client Certificates: %s", err)
-			return nil, err
-		}
-		cfg.Certificates = append(cfg.Certificates, cert)
+	// They provided SSL options, so let's load them.
+	blog.GetAuditLogger().Info("Loading SSL Options.")
+
+	if strings.HasPrefix(conf.AMQP.Server, "amqps") == false {
```

Comment:
Why not pull this check out to line 211? Is there some reason to allow non-AMQPS without some sort of explicit "I don't want security" confguration?